dist/scss/wds-components/_buttons.scss

```diff
@@ -98,6 +100,28 @@
 		}
 	}
 
+	&.wds-has-icon {
+		padding-left: $button-padding-horizontal + $wds-icon-small-size + 6px;
+		position: relative;
+
+		&.wds-is-squished {
+			padding-left: $button-squished-padding-horizontal + $wds-icon-tiny-size + 6px;
+
+			.wds-icon {
+				left: $button-squished-padding-horizontal;
+				margin-top: -$wds-icon-tiny-size / 2;
+			}
+		}
+
+		.wds-icon {
+			left: $button-padding-horizontal;
+			margin-top: -$wds-icon-small-size / 2;
+			pointer-events: none;
+			position: absolute;
```

Comment:
we can't because that causes another problems. And it's better to have all buttons same display value - `inline-block`